controllers/k8ssandra/datacenters.go

```diff
@@ -125,27 +131,47 @@ func (r *K8ssandraClusterReconciler) reconcileDatacenters(ctx context.Context, k
 				}
 			}
 
-			if !cassandra.DatacenterReady(actualDc) {
-				logger.Info("Waiting for datacenter to become ready")
+			if actualDc.Status.CassandraOperatorProgress != cassdcapi.ProgressReady {
+				logger.Info("Waiting for datacenter to finish updating")
 				return result.RequeueSoon(r.DefaultDelay), actualDcs
 			}
 
-			logger.Info("The datacenter is ready")
+			if actualDc.Spec.Stopped {
+				if !cassandra.DatacenterStopped(actualDc) {
+					logger.Info("Waiting for datacenter to satisfy Stopped condition")
+					return result.RequeueSoon(r.DefaultDelay), actualDcs
```

Comment:
> And if the datacenter is stopped, can't we just move on to the next one?

Here we are waiting for the Stopped condition to be true, since that's what we want. When it becomes true, we would move to the next dc.